proposals/0026/README.md

```diff
@@ -0,0 +1,275 @@
+
+In this proposal, a machine is not trusted any more than in Tinkerbell's API today.
+By design, in this proposal machines will have decreased access than they do today, with access to the APIs listed above.
+As it is in Tinkerbell today, the network is trusted when provisioning hardware.
+
+**Hegel**
+
+Hegel will only require read-only access to Hardware CRDs in Kubernetes.
+RBAC policies would govern the level of access to the Kubernetes API.
+
+Hegel will need to guard against confused-deputy attacks and not return metadata to the wrong client.
+
+**Boots**
+
+Boots will only require read-only access to Hardware CRDs in Kubernetes.
+RBAC policies would govern the level of access to the Kubernetes API.
+
+**PBnJ**
+
+PBnJ will require:
+* Read-only access to Hardware CRDs to discover management interface connectivity
+* Mutating access to Workflow CRDs to execute workflow steps like power cycling and BIOS management
+* Some level of secret access to connect to management interfaces.
+  Implementation and design of that access is out of scope of this proposal.
+
+**Tink API**
+
+The Tinkerbell API will require:
+* Read access to Hardware CRDs in Kubernetes
+* Read access to Template CRDs in Kubernetes
+* Write access to Workflow CRDs in Kubernetes
+
+**Tinkerbell Workflow Controller**
+
+There will need to be a controller to process creation of workflows.
+It will need:
+* Read access to Hardware CRDs in Kubernetes
+* Read access to Template CRDs in Kubernetes
+* Write access to Workflow CRDs `.spec` in Kubernetes
+
+**Tinkerbell Validating Webhook**
+
+A validating webhook will be used to validate Templates and Workflows as they are created.
+It will not require any Kubernetes API access, but needs to be reachable by the Kubernetes API server.
+
+**Tinkerbell Administrator**
+
+Tinkerbell administrators might require varying levels of access in order to perform CRUD operations on any or all Tinkerbell types.
+Kubernetes cluster administrators can define custom levels of access with RBAC policies to grant Tinkerbell admins the correct least-privilege level of access.
+
+## Alternatives
+
+The main alternative to achieve the stated design goal around availability would leave the Tinkerbell API alone, and leverage/modify the Tinkerbell server's [internal Database interface][tink-db-iface] to function on top of Kubernetes or a key-value datastore like [etcd][etcd].
+This alternative would help with high-availability deployments, but all the other motivations would remain unaddressed, and need to be implemented in Tinkerbell's API.
+
+Another alternative would be support the current architecture and codebase for existing functionality, and any new functionality would require migrating to use the Kubernetes resource model.
+This proposal does include maintianing the existing functionality and architecture for some period of time, but supporting it indefinitely would add significantly diverging code paths and increased maintenance effort.
```

Comment:
```suggestion
This proposal does include maintaining the existing functionality and architecture for some period of time, but supporting it indefinitely would add significantly diverging code paths and increased maintenance effort.
```

This is the answer to the TODO question above: https://github.com/tinkerbell/proposals/pull/46/files#diff-8f31e22643965ab377983da2fcba8b948c5a61c28744e999adb60c94d8e6fa1bR104

Perhaps replace the TODO with this line.